### PR TITLE
Us29 01 visitor trip summary

### PR DESCRIPTION
**Pull request by RaphaelBS-WCS**

### CODE REVIEW

**Review by jaldabaoth-code — DISMISSED** [review]

File: src/Service/FormatDuration.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            $$timeUnit = floor($time / $secondsPerUnit);
+            $time = $time % $secondsPerUnit;
+
+            if ($$timeUnit > 0 || !empty($result)) {
+                $result .= $$timeUnit . " $timeUnit ";
+            }
+        }
+        return trim($result);
+    }
+}
```
Comment (LucyMerlier) [inline src/Service/FormatDuration.php]: question : wouldn't it be more logical to manipulate DateInterval objects?

File: templates/includes/_tripSummary.html.twig [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<div class="visitorForm d-flex flex-column justify-content-center text-dark bg-light rounded-5 col-12 p-4">
+    <h2 class="text-warning text-center col-12 h1">Evaluation de votre trajet</h2>
+    <h3 class="h3 fw-light text-center">Vous faites en moyenne {{ app.request.get('tripSummary').distance }} kilomètres par jour,
+    soit {{ app.request.get('tripSummary').duration }} heures passés dans votre voiture.</h3>
+    <h3 class="h3 fw-light text-center"> Cela équivaut à<br>{{ app.request.get('tripSummary').annualDistance }} kilomètres<br>
```
Comment (LucyMerlier) [inline templates/includes/_tripSummary.html.twig:10]: "kilomètres * par an *"

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

File: templates/includes/_tripSummary.html.twig (hunk 2 above)
Comment (RaphaelBS-WCS) [reply templates/includes/_tripSummary.html.twig:10]: alright! :-)

File: src/Service/FormatDuration.php (hunk 1 above)
Comment (RaphaelBS-WCS) [reply src/Service/FormatDuration.php]: yes, done! :-)

**Review by jaldabaoth-code — APPROVED** [review]

**Review by LucyMerlier — APPROVED** [review]

File: src/Service/FormatDuration.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        $duration = $duration->format($day . ',' . $hour . ',' . '%i');
+
+        $durations = explode(',', $duration);
+        $result = [
+            'days' => $durations[0],
+            'hours' => $durations[1],
+            'minutes' => $durations[2]
+        ];
+
+        return $result;
```
Comment (LucyMerlier) [inline src/Service/FormatDuration.php:42]: the goal when using dateinterval was to return a dateinterval, not a string or array, and manage the display of the trip duration inside the view => here, you're already thinking about how you will display your duration(days, then hours, then minutes), but how to display stuff should always be the responsibility of the view

**Review by LucyMerlier — APPROVED** [review]

**Pull request merged**